src/libsyntax/ext/expand.rs

```diff
@@ -439,11 +440,13 @@ impl<'a, 'b> MacroExpander<'a, 'b> {
             }
             ProcMacroDerive(..) | BuiltinDerive(..) => {
                 self.cx.span_err(attr.span, &format!("`{}` is a derive mode", attr.path));
+                self.cx.trace_macros_diag();
                 kind.dummy(attr.span)
```

Comment:
I don't think we need `self.cx.trace_macros_diag();` when we're not panicking immediately after.